src/modelldcatnotordf/modelldcatno.py

```diff
@@ -1304,6 +1321,20 @@ def _has_simple_type_to_graph(self, _self: Any) -> None:
 
             self._g.add((_self, MODELLDCATNO.hasSimpleType, _has_simple_type))
 
+    def _has_data_type_to_graph(self, _self: Any) -> None:
+
+        if getattr(self, "has_data_type", None):
+
+            if getattr(self._has_data_type, "identifier", None):
+                _has_data_type = URIRef(self._has_data_type.identifier)
+            else:
+                _has_data_type = BNode()
```

Comment:
Samme som forrige PR:

```
_has_data_type = URIRef(self. _has_data_type.identifier) if hasattr(self. _has_data_type, "identifier") else BNode()
```

Reply:
Jeg prøvde dette i går, men dette gir meg en pyi eller pype error slik det er satt opp her hos meg :-( 

```
[1/20] check modelldcatnotordf.modelldcatno
FAILED: /home/frederik/code/fdk/modelldcatnotordf/.pytype/pyi/modelldcatnotordf/modelldcatno.pyi 
/home/frederik/code/fdk/modelldcatnotordf/.nox/pytype/bin/python -m pytype.single --disable import-error --imports_info /home/frederik/code/fdk/modelldcatnotordf/.pytype/imports/modelldcatnotordf.modelldcatno.imports --module-name modelldcatnotordf.modelldcatno -V 3.7 -o /home/frederik/code/fdk/modelldcatnotordf/.pytype/pyi/modelldcatnotordf/modelldcatno.pyi --analyze-annotated --nofail --quick /home/frederik/code/fdk/modelldcatnotordf/src/modelldcatnotordf/modelldcatno.py
File "/home/frederik/code/fdk/modelldcatnotordf/src/modelldcatnotordf/modelldcatno.py", line 275, in identifier: No attribute '_identifier' on None [attribute-error]
Called from (traceback):
  line 1330, in _has_data_type_to_graph
```

Reply:
Har du sette denne feilen før @azaxarov ?

Reply:
Nei, har ikke sett den før, men noe stemmer ikke. Jeg lurer på hva har `_identifier ` å gjøre med saken? Også ser jeg at jeg bommet på mellomrom etter `self`

Reply:
Kan du vise kodelinjen 275? Det høres ut som `var._identifier` der `var` er null

Reply:
Ja, dette er rart. Det var derfor det stoppet opp med å innføre denne kodeendringen som dere foreslo i går. Vi kan eventuelt vente og se om det er noen av de andre som har vært borti dette her før.

Reply:
Men jeg har to tasker på teknisk vedlikehold på å gjøre noe med dette, både det som du sa og det som German påpekte: 

https://github.com/Informasjonsforvaltning/modelldcatnotordf/issues/35

https://github.com/Informasjonsforvaltning/modelldcatnotordf/issues/32

Men foreløpig har jeg prioritert å få ut nye features før jeg tar ferie og andre eventuelt skal bruke parseren i juleferien. Men det er på ingen måte glemt, men noe av disse tingene tar tid å rette på siden det er ting som er gjennomført i hele koden.

Det kommer forresten en PR etter dette på en en teknisk sak, noe som German påpekte..

Reply:
God morgen @Chlenix , det som er null er i getteren til identifier:

```    @property
    def identifier(self) -> str:
        """Get for identifier."""
        return self._identifier
```

Altså, den siste linjen, return-statement'et

Reply:
Vel, vi kan ikke vente på at andre støter på det samme problem. Konseptuelt-sett er det ingenting feil med forslaget. I beste case, så avdekker det en mulig bug et annet sted i koden. Derfor mener jeg at vi bør forstå årsaken.

Reply:
Det som er litt ekstra med dette biblioteket at det er nokså heavy linting og mer ekstra regler tror jeg enn de andre bibliotekene. Men dette er jo også bra fordi det øker kodekvaliteten. Uansett. når dette skal fikses, så må det gjøres gjennomgående i hele koden. Og akkurat det er litt utenfor scope for akkurat denne tasken her.

Reply:
Følgande variant fungerer for meg. Den tar høgde for one-liner if, men brukar getattr:
```
_has_data_type = URIRef(self._has_data_type.identifier) if getattr(self._has_data_type, "identifier", None) else BNode()
```
Bruken av gettattr vs hasattr stammar frå datacatalogtordf, som Fredrik har tatt utgangspunkt i. Eg brukte _mykje_ tid på dette, og viss nokon kan vise meg korleis ein brukar hasattr _utan_ å ryke på pytype attribute-error, blir eg veldig happy. 

Ref også denne: https://hynek.me/articles/hasattr/

Reply:
> Ref også denne: https://hynek.me/articles/hasattr/

![image](https://user-images.githubusercontent.com/3957655/101890783-785bfd00-3ba1-11eb-9b06-f99a27ad7ee6.png)

Heldigvis bruker vi Python 3, og `hasattr` skal være bedre leselig enn getattr med 3 argumenter. Frederik får erroren fordi `self._has_data_type` er null, så `hasattr(self._has_data_type, 'identifier')` blir om til `"hasattr(null, 'identifier')"`